Получить все данные рекомендации

s3nkyh opened the pull request.

Commit:
added: RecomContrl, RecomDto, RecomService, SkillOfferDto

Commit:
--

Commit:
-

Commit:
--

Commit:
исправил RecommendationMapper

Commit:
убрал лишние аннотации

Commit:
исправил RecommendationMapper
добавил метод deleteAllAndCreate в RecommendationService

Commit:
реализовал механику удаления рекомендации

Commit:
реализовал механику получение всех данных рекомендации

Commit:
реализовал механику получения всех данных рекомендации

Commit:
добавил проверку на null

SergeyErmoshenko commented on src/main/java/school/faang/user_service/controller/RecommendationController.java:
```diff
@@ -0,0 +1,26 @@
+        return recommendationService.update(recommendationDto);
+    }
+
+    public void deleteRecommendation(Long id) {
```
Логов не хватает сюда и в сервис

SergeyErmoshenko commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,91 @@
+        Page<Recommendation> recommendationPage = recommendationRepository.findAllByAuthorId(authorId, pageable);
+        List<Recommendation> recommendationList = recommendationPage.getContent();
+        return recommendationList.stream()
+                .map(recommendationMapper::toRecommendationDto)
```
Лучше добавить метод в маппере, который преобразует срузу список

SergeyErmoshenko commented on src/main/java/school/faang/user_service/mapper/RecommendationMapper.java:
```diff
@@ -0,0 +1,53 @@
+
+@Mapper(componentModel = "spring")
+public interface RecommendationMapper {
+
```
строка пустая

SergeyErmoshenko commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,91 @@
+
+        skillOfferRepository.deleteAllByRecommendationId(recommendation.getId());
+        for (SkillOffer skillOffer : recommendation.getSkillOffers()) {
+            skillOfferRepository.create(skillOffer.getSkill().getId(), recommendation.getId());
```
Стоит подумать, может ли skill в recommendation быть null

SergeyErmoshenko commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,91 @@
+             throw new DataValidationException("Author id can't be null");
+        } else if (receiverId == null) {
+            throw new DataValidationException("Author id can't be null");
+        }
```
Еще можно проветить content на null

s3nkyh replied:
на него уже есть проверка

SergeyErmoshenko commented on src/main/java/school/faang/user_service/utils/ValidationRecommendationUtils.java:
```diff
@@ -0,0 +1,55 @@
+import java.time.Period;
+import java.util.List;
+
+public class ValidationRecommendationUtils {
```
Лучше переименовать в RecommendationValidation и перенести в папку utils/validationUtils

SergeyErmoshenko commented on src/main/java/school/faang/user_service/utils/ValidationRecommendationUtils.java:
```diff
@@ -0,0 +1,55 @@
+
+    public static void validateRecommendationContent(RecommendationDto recommendationDto) {
+        if (recommendationDto.getContent() == null) {
+            throw new DataValidationException(CONTENT_NULL_EXCEPTION);
```
Лучше перед выбросом ошибки добавить еще лог такой же

SergeyErmoshenko commented on src/main/java/school/faang/user_service/utils/ValidationRecommendationUtils.java:
```diff
@@ -0,0 +1,55 @@
+        LocalDate currentDate = LocalDate.now();
+        Period period = Period.between(currentDate, recommendationDto.getCreatedAt().toLocalDate());
+
+        if (period.toTotalMonths() <= 6) {
```
6 лучше вынести в константу

SergeyErmoshenko commented on src/main/java/school/faang/user_service/utils/ValidationRecommendationUtils.java:
```diff
@@ -0,0 +1,55 @@
+
+    public static void validateSkills(RecommendationDto recommendationDto, List<SkillOffer> allSkillOffers) {
+        List<SkillOfferDto> skillOfferDtoList = recommendationDto.getSkillOffersDto();
+        List<Long> allSkillOffersIds = allSkillOffers.stream()
```
тут тоже маппер использовать сразу в лист и можно преобразовать в HashSet

SergeyErmoshenko reviewed (changes requested):
тк ветка была создана до того, как в мастер запушили gitignore логов, удали их из PR

Commit:
исправил RecommendationService,
RecommendationValidation,
RecommendationMapper
добавил метод в RecommendationRepository и SkillOfferRepository
